From 182220e26134122e6bf12ea24c68f1a022c8da9e Mon Sep 17 00:00:00 2001
From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 8 Apr 2010 23:38:56 +0200
Subject: PM / Hibernate: user.c, fix SNAPSHOT_SET_SWAP_AREA handling
References: bko#15728
Patch-mainline: on-the-way

When CONFIG_DEBUG_BLOCK_EXT_DEVT is set we decode the device
improperly by old_decode_dev and it results in an error while
hibernating with s2disk.

All users already passes the new device number, so switch to
new_decode_dev.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Cc: "Rafael J. Wysocki" <rjw@sisk.pl>
Cc: Pavel Machek <pavel@ucw.cz>
---
 kernel/power/user.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/kernel/power/user.c b/kernel/power/user.c
index 9eafb8c..6ca76a6 100644
--- a/kernel/power/user.c
+++ b/kernel/power/user.c
@@ -593,7 +593,7 @@ static long snapshot_ioctl(struct file *filp, unsigned int cmd,
 			 * User space encodes device types as two-byte values,
 			 * so we need to recode them
 			 */
-			swdev = old_decode_dev(swap_area.dev);
+			swdev = new_decode_dev(swap_area.dev);
 			if (swdev) {
 				offset = swap_area.offset;
 				data->swap = swap_type_of(swdev, offset, NULL);
-- 
1.7.0.3

